### ISSUE TITLE
Regression with nav item direct linking to blog post in 4.40.3

### ISSUE THREAD

**Issue by perpil** [squidfunk/mkdocs-material#5949]
### Context

In a recent change, directly linking to a blog post from the nav exhibits strange behavior.  The nav link works, but that specific blog post is missing the metadata sidebar and it is excluded from the blog index.

### Bug description

On my [site](https://speedrun.nobackspacecrew.com/), I link to a specific blog entry under a nav entry called **Mission**.  This normally works fine, but in insiders 9.2.7+4.40.3 I get an error: `A reference to 'blog/posts/mission.md' is included in the 'nav' configuration, but this file is excluded from the built site.` The nav link works, and it shows the blog entry, but it is missing the blog metadata like date and author.  Other posts that aren't linked in the nav show up in the blog index, but the `mission.md` one is missing.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- #5904

### Reproduction

[9.2.7+insiders.4.40.3-blog-entry-in-nav-not-working.zip](https://github.com/squidfunk/mkdocs-material/files/12505125/9.2.7%2Binsiders.4.40.3-blog-entry-in-nav-not-working.zip)

### Steps to reproduce

1. mkdocs serve
2. See the error `A reference to 'blog/posts/mission.md' is included in the 'nav' configuration, but this file is excluded from the built site.`
3. Click the `Mission` nav item, note it does not have metadata
4. Click the `Blog` nav item, note there is no entry for the mission blog post.
5. Click `continue reading` and note the post called Entry has metadata.

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. What was the last version working? I suspect that this is caused by the latest refactoring of the plugin, which was necessary to make it work with other third-party plugins like awesome-pages or static-i18n. The blog posts must be temporarily excluded, because, otherwise, MkDocs will add all of them to the navigation. This is unusual for blogs, because blog posts are usually linked from paginated indexes.

We already filed an upstream issue 3 weeks ago, so we could mark posts as "not in nav", but not "excluded":

- https://github.com/mkdocs/mkdocs/issues/3336

Until the upstream issue is resolved, fixing the problem reported is blocked, because we currently have to hack around it.

**Comment by squidfunk** [comment]
Okay, I have an even better idea now. Fixing the issue mentioned in my last comment will fix the warning that is printed. However, the problem that the page is overwritten stems from the fact that MkDocs will ignore that pages are excluded from the navigation when they are linked in the navigation, and always create (= overwrite in case of blog posts) them. In fact, if the following PR that I proposed several days ago gets merged and release, the problem is gone:

- https://github.com/mkdocs/mkdocs/pull/3367
- https://github.com/mkdocs/mkdocs/pull/3366

**Comment by squidfunk** [comment]
Fixed in 3f87fe0f5. I've extended the temporary hack that works around the limitations mentioned in https://github.com/mkdocs/mkdocs/issues/3366 and fixed by https://github.com/mkdocs/mkdocs/issues/3367 which we already use for the blog entrypoint. Applying them to posts and views fixes the problem for them as well. Once the mentioned PR is merged, the entire part of the code can be removed.

**Comment by perpil** [comment]
Thanks for the quick fix! I verified it works, but the nav uses the title of the blog post instead of the nav key.  I've attached a repro demonstrating this.  Is this change in behavior intentional?

In this repro, the nav previously would be set to `Mission`, now it is set to `The Mission Statement`

[9.2.7+insiders.4.40.3-nav-using-title.zip](https://github.com/squidfunk/mkdocs-material/files/12506471/9.2.7%2Binsiders.4.40.3-nav-using-title.zip)

Also per your question above about the last known version this was working. According to the metadata in my published site it was working with: `mkdocs-1.5.2, mkdocs-material-9.2.3+insiders-4.39.3`

**Comment by squidfunk** [comment]
Thanks again for the feedback and reproduction.

> Also per your question above about the last known version this was working. According to the metadata in my published site it was working with: mkdocs-1.5.2, mkdocs-material-9.2.3+insiders-4.39.3

If you're referring to the post title being set to the nav key – that is weird, because posts never had explicit titles set as they are not meant to be linked in the navigation, so the fact that this worked before was definitely accidental and not intended. The problem is: we can't allow to link posts in the navigation, because we must process posts _before_ the navigation is constructed because several parts of the blog plugin rely on this information.

This is orthogonal to what you're trying to achieve and the reason why the behavior is not in line what you'd expect from MkDocs. Once https://github.com/mkdocs/mkdocs/pull/3367 is merged, post titles can't be set at all, and IMHO that is correct. Thus, we're not considering this a bug but undefined behavior as you're outside of what is supported by the blog plugin.

You can work around this restriction by using a relative link:

``` yaml
nav:
  - Blog:
    - blog/index.md 
  - Mission: blog/2023/09/01/the-mission-statement/
validation:
  nav:
    not_found: ignore
```

**Comment by perpil** [comment]
I thought the key in the nav would set the nav text, so I was a bit surprised that it was resolving to the destination title, but the workaround works for me.  Thanks!

**Comment by squidfunk** [comment]
Yes, it sets the nav text, for normal pages. Blog posts are not normal pages, they are excluded from the navigation, because that's usually not how blogs work ☺️ We'll clarify this better in our docs. Glad the workaround works for you!

**Comment by squidfunk** [comment]
Released as part of [9.2.8](https://squidfunk.github.io/mkdocs-material/changelog/#9.2.8).
